apps/site/src/app/(main)/mentor/Form/Form.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+
+import { FormEvent, useState } from "react";
+import axios from "axios";
+
+import Textfield from "@/lib/components/forms/Textfield";
+import Button from "@/lib/components/Button/Button";
+
+import AgeInformation from "@/lib/components/forms/shared/AgeInformation";
+import SchoolInformation from "@/lib/components/forms/shared/SchoolInformation";
+import ResumeInformation from "@/lib/components/forms/shared/ResumeInformation";
+import BasicInformation from "./BasicInformation";
+import ProfileInformation from "./ProfileInformation";
+import ShortAnswers from "./ShortAnswers";
+import ExperienceInformation from "./ExperienceInformation";
+
+import hasDeadlinePassed from "@/lib/utils/hasDeadlinePassed";
+import RadioSelect from "@/lib/components/forms/RadioSelect";
+
+import styles from "@/lib/components/forms/shared/Form.module.scss";
+
+const APPLY_PATH = "/api/user/apply";
+const FIELDS_WITH_OTHER = [
+	"pronouns",
+	"ethnicity",
+	"school",
+	"major",
+	"other_mentor_experience",
+];
+const SELECT_OPTIONS = [
+	"frontend",
+	"backend",
+	"mobile",
+	"databases",
+	"ai/ml",
+	"vr",
+	"blockchain",
+	"embedded",
+	"data_science",
+	"cybersecurity",
+];
+
+export default function Form() {
+	const [submitting, setSubmitting] = useState(false);
+	const [sessionExpired, setSessionExpired] = useState(false);
+
+	const handleSubmit = async (
+		event: FormEvent<HTMLFormElement>,
+	): Promise<void> => {
+		// Disable native post submission
+		event.preventDefault();
+
+		// On the chance that a user presses submit after the deadline,
+		// this check is to prevent the application from submitting
+		// and to show the message that applications have closed
+		if (hasDeadlinePassed()) {
+			window.location.reload();
+			return;
+		}
+
+		setSubmitting(true);
+		setSessionExpired(false);
+
+		const formData = new FormData(event.currentTarget);
+
+		// Use other values when selected
+		for (const field of FIELDS_WITH_OTHER) {
```

Comment:
Suggestion: if we want to use the common Form without refactoring further, we can move this logic into the component in which it originates and update the input attributes there when other is selected instead of at submission time